### PR TITLE
Unpin GRPC dependencies

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup
/kind dependency-change

#### What this PR does / why we need it:
This removes the pinned dependencies to stick to the latest versions.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
Shortly discussed with @fidencio via slack, whereas we think we're good to go with this change and kata.
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@fidencio now kata seems to fail again :thinking:

**Comment by fidencio** [comment]
> @fidencio now kata seems to fail again thinking

Let me investigate why it's failing ...

**Comment by fidencio** [comment]
So, this is totally related to the gogoprotobuf issues we faced in the past.
```
panic: protobuf tag not enough fields in Status.state:
goroutine 246 [running]:
github.com/gogo/protobuf/proto.(*unmarshalInfo).computeUnmarshalInfo(0xc0003d6aa0)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/table_unmarshal.go:341 +0x1834
github.com/gogo/protobuf/proto.(*unmarshalInfo).unmarshal(0xc0003d6aa0, 0xc0002dd980, 0xc000c2e004, 0x0, 0x7, 0x0, 0x27f2040)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/table_unmarshal.go:138 +0xe02
github.com/gogo/protobuf/proto.makeUnmarshalMessagePtr.func1(0xc000c2e004, 0x7, 0x8, 0xc000527a40, 0x2, 0x7f1e0c5ac870, 0x7f1e0e8eafff, 0x7f1e0c08afc8, 0xc000078b20, 0x4196a6)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/table_unmarshal.go:1826 +0x12b
github.com/gogo/protobuf/proto.(*unmarshalInfo).unmarshal(0xc0003d6640, 0xc000527a40, 0xc000c2e002, 0x9, 0x9, 0x419f18, 0x20)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/table_unmarshal.go:175 +0x79e
github.com/gogo/protobuf/proto.(*InternalMessageInfo).Unmarshal(0xc0008be000, 0x2796d80, 0xc000527a40, 0xc000c2e002, 0x9, 0x9, 0x1, 0x0)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/table_unmarshal.go:63 +0x66
github.com/gogo/protobuf/proto.(*Buffer).Unmarshal(0xc000371cf0, 0x2796d80, 0xc000527a40, 0x0, 0x0)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/decode.go:424 +0x1ec
github.com/gogo/protobuf/proto.Unmarshal(0xc000c2e002, 0x9, 0x9, 0x2796d80, 0xc000527a40, 0xc0006ed638, 0xc000078f90)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/gogo/protobuf/proto/decode.go:342 +0x170
github.com/containerd/ttrpc.(*Client).recv(0xc0005f3500, 0xc000527a40, 0xc000674180, 0x0, 0x0)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/containerd/ttrpc/client.go:337 +0x14f
github.com/containerd/ttrpc.(*Client).run(0xc0005f3500)
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/containerd/ttrpc/client.go:287 +0x471
created by github.com/containerd/ttrpc.NewClient
        /home/fidencio/go/src/github.com/cri-o/cri-o/vendor/github.com/containerd/ttrpc/client.go:94 +0x1bd

```

Interestingly enough, I've tested #4883, which bumped protobuf to v1.5.1 and  grpc to v1.37.0 and there was no issue at all.

I wonder whether bumping to a newer version would be okay to the CRI-O team, such as done in the PR mentioned above.

/cc @fgiudici, as he proposed on a fix on containerd side, which would solve the problem once and for all.

**Comment by fidencio** [comment]
LGTM, thanks @saschagrunert!

I've tested this bump locally and it does work as expected.

**Comment by fidencio** [comment]
I see it's still failing, let me re-run the CI while I can debug it locally as well.

File: go.mod [hunk 1]
```diff
@@ -70,13 +70,12 @@ require (
 )
 
 replace (
-	github.com/golang/protobuf => github.com/golang/protobuf v1.3.5
+	github.com/golang/protobuf => github.com/golang/protobuf v1.5.1
 	github.com/opencontainers/runtime-spec => github.com/opencontainers/runtime-spec v1.0.3-0.20201121164853-7413a7f753e1
 	// Pinning the syndtr/gocapability until https://github.com/opencontainers/runc/commit/6dfbe9b80707b1ca188255e8def15263348e0f9a
 	// is included in the runc release
 	github.com/syndtr/gocapability => github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2
-	google.golang.org/genproto => google.golang.org/genproto v0.0.0-20200117163144-32f20d992d24
```
Comment (fidencio) [inline go.mod:78]: @saschagrunert, okay, seems that I got the main difference that's causing the errors we're facing in the CI.
Some time ago when I tested this I didn't remove genproto, and on local tests seems that just by keeping genproto pinned is enough to have kata-containers happy, even if we remove the pinned protobuf and grpc.

Do you think that leaving genproto pinned could cause any issue?  If no, please, just update the PR to remove the pin on both protobuf & grpc, leaving the genproto pinned, and we'd be good to go.

If it cause some issue on the CRI-O side, what's the issue caused?

Please, see https://github.com/cri-o/cri-o/pull/4957, which used a still pinned version but didn't remove genproto. (CI: http://jenkins.katacontainers.io/job/kata-containers-2-crio-PR/168/console)

**Comment by haircommander** [comment]
should we un-WIP this?

**Comment by saschagrunert** [comment]
> should we un-WIP this?

Yep, CI is green now. :+1:

**Pull request merged**